### PR TITLE
[Pal/Linux-SGX] Skip "TOML parsing" message in child enclaves

### PR DESCRIPTION
**Pull request by dimakuv**
## Description of the changes 

The "TOML parsing" message is always printed on enclave startup (because we cannot rely on the `log_level` specified in the manifest at startup). Previously, it was annoyingly printed for every child enclave.

## How to test this PR? 

An example of R, that calls `rm` and <some other utility> upon termination:

- before this PR:
```
~/gramineproject/examples/r$ gramine-sgx ./R --slave --vanilla -f scripts/sample.r
Parsing TOML manifest file, this may take some time...
-----------------------------------------------------------------------------------------------------------------------
Gramine detected the following insecure configurations:

  - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
  - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)

Gramine will continue application execution, but this configuration must not be used in production!
-----------------------------------------------------------------------------------------------------------------------

[P1:T1:] error: Detected deprecated syntax: 'fs.mount'. Consider converting to the new array syntax: 'fs.mounts = [{ type = "chroot", uri = "...", path = "..." }]'.
[1] 3 2 1
[1] 3 1 2
[1] 3 5 2 5
 [1] 2 2 2 2 1 2 2 2 2 2
[1] "success"
Parsing TOML manifest file, this may take some time...
Parsing TOML manifest file, this may take some time...
```

- with this PR:
```
~/gramineproject/examples/r$ gramine-sgx ./R --slave --vanilla -f scripts/sample.r
Gramine starts. Parsing TOML manifest file, this may take some time...
-----------------------------------------------------------------------------------------------------------------------
Gramine detected the following insecure configurations:

  - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
  - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)

Gramine will continue application execution, but this configuration must not be used in production!
-----------------------------------------------------------------------------------------------------------------------

[P1:T1:] error: Detected deprecated syntax: 'fs.mount'. Consider converting to the new array syntax: 'fs.mounts = [{ type = "chroot", uri = "...", path = "..." }]'.
[1] 3 2 1
[1] 3 1 2
[1] 2 3 5 5
 [1] 2 2 2 2 2 2 2 2 1 2
[1] "success"
```

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)

### CODE REVIEW

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: 0 of 1 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 927 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2QTRsCOXm4sbd1tUr:-N-2QTRsCOXm4sbd1tUs:b-hnwl78) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L927)):*
> ```C
>     if (parent_stream_fd < 0) {
>         /* only print during main process's startup (note that this message is always printed) */
>         log_always("Gramine starts. Parsing TOML manifest file, this may take some time...");
> ```

FYI: I added this `Gramine starts` prefix to make it more "mild" for unprepared new users of Gramine. Since this message is always printed, it is kinda scary for new users who don't know what TOML is, and what's going on.

**Review by boryspoplawski — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @dimakuv)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2arKv1FMneArZADSD:bd75e0y) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
> ```C
>         return -EINVAL;
>     }
>     log_warning("Gramine parsed TOML manifest file successfully");
> ```

Actually, why this is a warning?

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @boryspoplawski)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2daOcE-TtE2juI1Yq:b-cwzil5) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

Actually, why this is a warning?
</blockquote></details>

This message is not supposed to be a logical warning, but we lack `log_info()`.

The log levels are:
- Always: but we don't want to print these info messages always
- Error: well this is not an error for sure
- Warning: chose this level since there is no better alternative
- Debug: maybe it should be debug? could be changed to debug, I don't see any reason why not

**Review by boryspoplawski — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @dimakuv)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2gqbi-pEbITkOqSyz:bxiervi) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

This message is not supposed to be a logical warning, but we lack `log_info()`.

The log levels are:
- Always: but we don't want to print these info messages always
- Error: well this is not an error for sure
- Warning: chose this level since there is no better alternative
- Debug: maybe it should be debug? could be changed to debug, I don't see any reason why not
</blockquote></details>

Yeah, I was just asking about why this level. This sounds like a debug message to me...
Anyway, unblocking as it was like this before.

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow, @pwmarcz, and @woju)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2jCtt6IRUc9tmPvRn:bsf9tp0) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

Yeah, I was just asking about why this level. This sounds like a debug message to me...
Anyway, unblocking as it was like this before.
</blockquote></details>

Actually, you are right. Let me block on this and ask other maintainers what they think.

@pwmarcz @mkow @woju What do you think about changing this (and similar) info messages from `log_warning()` to `log_debug()`? Note couple things:
- `log_warning` is not supposed to be used in production, so it is no better than using `log_debug` in the security sense
- our users typically choose between `error`, `debug` and `trace` levels. I think I've never seen anyone using `warning` :) So dropping from `warning` to `debug` won't affect most (99%?) of our users

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow, @pwmarcz, @woju, and @ying2liu)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2kfHLB0rJ_rpB34LB:b-dam3oq) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Actually, you are right. Let me block on this and ask other maintainers what they think.

@pwmarcz @mkow @woju What do you think about changing this (and similar) info messages from `log_warning()` to `log_debug()`? Note couple things:
- `log_warning` is not supposed to be used in production, so it is no better than using `log_debug` in the security sense
- our users typically choose between `error`, `debug` and `trace` levels. I think I've never seen anyone using `warning` :) So dropping from `warning` to `debug` won't affect most (99%?) of our users
</blockquote></details>

@ying2liu As you introduced these messages, was there a specific reason to choose `log_warning()` instead of `log_debug()`? Do you think we can change these messages to `log_debug()`?

**Review by pwmarcz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow, @woju, and @ying2liu)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-4847nBP8M8yO6-uls:b7uy55s) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@ying2liu As you introduced these messages, was there a specific reason to choose `log_warning()` instead of `log_debug()`? Do you think we can change these messages to `log_debug()`?
</blockquote></details>

 I'd downgrade this to `debug`. According to [how we describe these levels in documentation](https://gramine.readthedocs.io/en/latest/manifest-syntax.html#common-syntax), `debug` contains information about Gramine internals, and that includes stages of initialization like loading the manifest.

Of course this might not matter too much to users, if they never set their log level to `warning` specifically. But that's what fits our scheme, IMO.

**Comment by ying2liu** [comment]
> > _[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)_ status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow, @woju, and @ying2liu)
> 
> _[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-4847nBP8M8yO6-uls:b7uy55s) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):_
> 
> Previously, dimakuv (Dmitrii Kuvaiskii) wrote…
> I'd downgrade this to `debug`. According to [how we describe these levels in documentation](https://gramine.readthedocs.io/en/latest/manifest-syntax.html#common-syntax), `debug` contains information about Gramine internals, and that includes stages of initialization like loading the manifest.
> 
> Of course this might not matter too much to users, if they never set their log level to `warning` specifically. But that's what fits our scheme, IMO.

At this point, log_level has not been defined yet.

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required) (waiting on @dimakuv and @woju)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 927 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-5UXb12-pwBA0Q11S3:-N-5UXb12-pwBA0Q11S4:undefined) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L927)):*
> ```C
>     if (parent_stream_fd < 0) {
>         /* only print during main process's startup (note that this message is always printed) */
>         log_always("Gramine starts. Parsing TOML manifest file, this may take some time...");
> ```





_Suggestion:_
```C
Gramine is starting.
```
___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 927 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2QTRsCOXm4sbd1tUr:-N-5UeejB9d2Dn9joS8P:bdc1tvr) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L927)):*
> new users who don't know what TOML is, and what's going on.

They need to already know this, as they had to configure the manifest to run Gramine, which is in TOML. But this version is also fine with me.
___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-5V1ht5Aangm7GkwwA:b-fd34vc) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, ying2liu (Ying Liu) wrote…</i></summary><blockquote>

At this point, log_level has not been defined yet.
</blockquote></details>

+1 to `log_debug`

**Comment by ying2liu** [comment]
> > _[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)_ status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow, @pwmarcz, @woju, and @ying2liu)
> 
> _[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-2kfHLB0rJ_rpB34LB:b-dam3oq) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):_
> 
> Previously, dimakuv (Dmitrii Kuvaiskii) wrote…
> @ying2liu As you introduced these messages, was there a specific reason to choose `log_warning()` instead of `log_debug()`? Do you think we can change these messages to `log_debug()`?

The reason why we added the message is that some customers complained that they would not get any response from Gramine for some time during parse_loader_config. If we change it to log_debug, the customers still could not get any response when using release build.

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required) (waiting on @dimakuv and @woju)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-70UGV5BsSCP8LbjRR:bhtj5la) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, ying2liu (Ying Liu) wrote…</i></summary><blockquote>

The reason why we added the message is that some customers complained that they would not get any response from Gramine for some time during parse_loader_config. If we change it to log_debug, the customers still could not get any response when using release build.
</blockquote></details>

@ying2liu But these messages are printed using `log_warning()`, which is *not* a proper release-version log level anyway (the only proper release-version log level is `error`). So in the release build (not actually the build, just the `log_level = "error"` option in the manifest) even current messages won't be printed. Or am I missing something?

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required) (waiting on @dimakuv and @woju)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-MQKm-AfU-m1kXC_Nv:b2rw8mm) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, ying2liu (Ying Liu) wrote…</i></summary><blockquote>

Sorry, I forgot some details. The purpose of using log_warning is to warn use in order to avoid any surprise. We can't use log_error, because that will mislead users. 
</blockquote></details>

@ying2liu But can we use `log_debug()` instead of `log_warning()`? Is there any particular objection to `log_debug()`?

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: 0 of 2 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @boryspoplawski and @mkow)

___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 927 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-5UXb12-pwBA0Q11S3:-N-WhBDa2e82jxET6nJ4:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L927)):*
> ```C
>     if (parent_stream_fd < 0) {
>         /* only print during main process's startup (note that this message is always printed) */
>         log_always("Gramine starts. Parsing TOML manifest file, this may take some time...");
> ```

Done.
___
*[Pal/src/host/Linux-SGX/sgx_main.c, line 934 at r1](https://reviewable.io/reviews/gramineproject/gramine/501#-N-2arKv1FMneArZADSC:-N-WhZ5Y0afQAoY1N1JC:bbodnm8) ([raw file](https://github.com/gramineproject/gramine/blob/bea4a8fb2e2cb5c2f122126ca562c47ae63050c0/Pal/src/host/Linux-SGX/sgx_main.c#L934)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@ying2liu But can we use `log_debug()` instead of `log_warning()`? Is there any particular objection to `log_debug()`?
</blockquote></details>

Done. I changed to `log_debug()`.

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, 1 unresolved discussion, "fixup! " found in commit messages' one-liners (waiting on @mkow)

**Review by mkow — DISMISSED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: all files reviewed, all discussions resolved, "fixup! " found in commit messages' one-liners

**Review by boryspoplawski — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/501)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**
